src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -80,6 +88,162 @@ private PackageArchiveReader BuildPackage(string path)
             }
         }
 
+        private PackageArchiveReader BuildFromProjectJson(string path)
+        {
+            PackageBuilder packageBuilder = CreatePackageBuilderFromProjectJson(path, _packArgs.GetPropertyValue);
+
+            if (_packArgs.Symbols)
+            {
+                // remove source related files when building the lib package
+                ExcludeFilesForLibPackage(packageBuilder.Files);
+
+                if (!packageBuilder.Files.Any())
+                {
+                    throw new ArgumentException(String.Format(CultureInfo.CurrentCulture, Strings.Error_PackageCommandNoFilesForLibPackage, path, Strings.NuGetDocs));
+                }
+            }
+
+            PackageArchiveReader package = BuildPackage(packageBuilder);
+
+            if (_packArgs.Symbols)
+            {
+                BuildSymbolsPackage(path);
+            }
+
+            if (package != null && !_packArgs.NoPackageAnalysis)
+            {
+                AnalyzePackage(package, packageBuilder);
+            }
+
+            return package;
+        }
+
+        private PackageBuilder CreatePackageBuilderFromProjectJson(string path, Func<string, string> propertyProvider)
+        {
+            // Set the version property if the flag is set
+            if (!String.IsNullOrEmpty(_packArgs.Version))
+            {
+                _packArgs.Properties["version"] = _packArgs.Version;
+            }
+
+            PackageBuilder builder = new PackageBuilder();
+
+            using (var stream = new FileStream(path, FileMode.Open, FileAccess.Read))
+            {
+                LoadProjectJsonFile(builder, path, _packArgs.BasePath, Path.GetFileName(Path.GetDirectoryName(path)), stream, propertyProvider);
+            }
+
+            return builder;
+        }
+
+        public static bool ProcessProjectJsonFile(PackageBuilder builder, string basePath, string id, Func<string, string> propertyProvider)
+        {
+            if (basePath == null)
+            {
+                return false;
+            }
+            string path = Path.Combine(basePath, "project.json");
+            if (File.Exists(path))
+            {
+                using (var stream = new FileStream(path, FileMode.Open, FileAccess.Read))
+                {
+                    LoadProjectJsonFile(builder, path, basePath, id, stream, propertyProvider);
+                }
+                return true;
+            }
+
+            return false;
+        }
+
+        private static void LoadProjectJsonFile(PackageBuilder builder, string path, string basePath, string id, Stream stream, Func<string, string> propertyProvider)
+        {
+            string content = Preprocessor.Process(stream, propName => propertyProvider(propName));
```

Comment:
This would be great, but it should be removed until we can get agreement on this.